src/algebra/algebra/spectrum.lean

```diff
@@ -0,0 +1,183 @@
+  is_unit (r • 1 - a) ↔ is_unit (1 - r⁻¹ • a) :=
+begin
+  have with_smul_one : is_unit (r•(1 : R)•1 - a) ↔ is_unit ((1 : R)•1 - r⁻¹•a),
+    by exact is_unit_smul_smul_sub_iff_is_unit_smul_sub_smul,
+  simp at with_smul_one,
+  exact with_smul_one,
+end
+
+lemma add_mem_spectrum {a : A} {r s : R} :
+  r ∈ (σ a : set R) ↔ r + s ∈ (σ (s • 1 + a) : set R) :=
+begin
+  apply not_iff_not.mpr,
+  change is_unit (r•1 - a) ↔ is_unit ((r+s)•1 - (s•1 + a)),
+  have h_eq : (r+s)•1 - (s•1 + a) = r•1 - a,
+    by { rw add_smul, noncomm_ring },
+  simp [h_eq],
+end
+
+lemma mul_mem_spectrum_mul {a : A} {s : R} {r : units R} :
+  r • s ∈ (σ (r • a) : set R) ↔ s ∈ (σ a : set R) :=
+begin
+  apply not_iff_not.mpr,
+  change is_unit ((r•s)•1 - r•a) ↔ is_unit (s•1 - a),
+  have h_eq : (r•s)•(1 : A) = r•s•1, by simp,
+  rw h_eq,
+  exact is_unit_smul_smul_sub_smul_iff,
+end
+
+theorem left_add_coset_spectrum (a : A) (r : R) :
+  left_add_coset r (σ a) = (σ (r • 1 + a)) :=
+by { ext, rw [mem_left_add_coset_iff, neg_add_eq_sub, add_mem_spectrum],
+     nth_rewrite 1 ←sub_add_cancel x r, }
+
+-- this one isn't quite as simple because `R` is not a `group` (but it is an `add_group`)
+-- therefore we don't have access to `mem_left_add_coset_iff`
+theorem unit_left_coset_spectrum (a : A) (r : units R) :
+  left_coset ↑r (σ a : set R) = σ (r • a) :=
+begin
+  ext,
+  split,
+  { rintro ⟨y,y_mem,hy⟩,
+    rw ←hy,
+    exact mul_mem_spectrum_mul.mpr y_mem, },
+  { intro hx,
+    have self_inv : x = r•r⁻¹•x, by simp,
+    rw self_inv at *,
+    exact mem_left_coset ↑r (mul_mem_spectrum_mul.mp hx), },
+end
+
+-- general (noncommutative) ring results about units and the identity
+lemma right_inv_of_right_inv_swap {a b c : A} (h : (1 - a * b) * c = 1) :
+  (1 - b * a) * (1 + b * c * a) = 1 :=
+calc (1 - b*a)*(1 + b*c*a) = 1 - b*a + b*((1 - a*b)*c)*a : by noncomm_ring
+...                        = 1                           : by simp [h]
+
+lemma left_inv_of_left_inv_swap {a b c : A} (h : c * (1 - a * b) = 1) :
+  (1 + b * c * a) * (1 - b * a) = 1 :=
+calc (1 + b*c*a)*(1 - b*a) = 1 - b*a + b*(c*(1 - a*b))*a : by noncomm_ring
+...                        = 1                           : by simp [h]
+
+lemma is_unit_one_sub_mul_of_swap {a b : A} (h : is_unit (1 - a * b)) :
+  is_unit (1 - b * a) :=
+by { let h₁ := right_inv_of_right_inv_swap h.unit.val_inv,
+     let h₂ := left_inv_of_left_inv_swap h.unit.inv_val,
+     exact ⟨⟨1-b*a,1+b*h.unit.inv*a,h₁,h₂⟩,rfl⟩, }
+
+lemma is_unit_one_sub_mul_iff_swap {a b : A} :
+  is_unit (1 - a * b) ↔ is_unit (1 - b * a) :=
+by { split, repeat {apply is_unit_one_sub_mul_of_swap}, }
+
+-- `r ∈ σ(a*b) ↔ r ∈ σ(b*a)` for any `r : units R`
+theorem unit_mem_spectrum_mul_iff_swap_mul {a b : A} {r : units R} :
+  ↑r ∈ (σ (a * b) : set R) ↔ ↑r ∈ (σ (b * a) : set R) :=
+begin
+  apply not_iff_not.mpr,
+  change is_unit (r•1 - a*b) ↔ is_unit (r•1 - b*a),
+  repeat {rw [is_unit_smul_sub_iff_is_unit_sub_smul]},
```

Comment:
You can probably replace repeat rw by `simp only` -- that's basically what it does.